s2s/time.py

```diff
@@ -170,6 +174,53 @@ def map_years(
 
         return index
 
+    def map_to_data(self, input_data: Union[pd.Series, pd.DataFrame, xr.Dataset, xr.DataArray],
+        flat: bool = False
+        ) -> pd.DataFrame:
+        """Map the calendar to input data period.
+        
+        Get datetime range from input data and generate correpodning interval index. This method
+        guarentess that the generated interval (calendar) indices would be covered by the input
+        data.
+
+        Args:
+            input_data: Input data for datetime mapping. Its index must be pandas.DatetimeIndex.
+            flat: Same as the argument in ``map_years``.
+
+        Returns:
+            Pandas DataFrame filled with Intervals of the calendar's frequency.
+            (also see ``map_year`` and ``map_years``)
+        """
+        # check the datetime order of input data
+        first_timestamp = input_data.index[0]
+        last_timestamp = input_data.index[-1]
```

Comment:
I was wondering if we could just use `input_data.index.max()` and `input_data.index.min()` here. This gives the maximum and minimum datetime values, and does not care about the order.

Reply:
Thanks for the comment! Indeed I just checked the function and the way you suggest basically can work with input data with backward datetime index as well. Super cool idea!